dll/cpl/desk/settings.c

```diff
@@ -842,6 +844,43 @@ SwitchDisplayMode(HWND hwndDlg, PWSTR DeviceName, PSETTINGS_ENTRY seInit, PSETTI
     }
 }
 
+static
+PSETTINGS_ENTRY
+FindBestElement(
+    _In_ PSETTINGS_ENTRY pInitialSettings,
+    _In_ PSETTINGS_ENTRY Current)
+{
+    LONG Penalty, SmallestPenalty = MAXLONG;
+    PSETTINGS_ENTRY pBestEntry = NULL;
+
+    /* Go back to 1st element */
+    while (Current->Blink)
+        Current = Current->Blink;
```

Comment:
I wonder whether giving to the FindBestElement function also the head of this linked list wouldn't be better.

Reply:
The head node is used. Thanks.